Release for GHC 9.6

Hi @jfischoff, I'm slowing rolling out GHC 9.6-rc1 to my projects, and while it seems like the repository HEAD of `postgres-options` is compatible with GHC 9.6 (or at least does not emit build errors), the latest version on Hackage has some problems:

```

src/Database/PostgreSQL/Simple/Options.hs:204:22: error: [GHC-39999]
Error: cabal: Failed to build postgres-options-0.2.0.0 (which is required by
test:entity-test from pg-entity-0.0.4.2). See the build log above for details.

Error:     • No instance for ‘Alternative (Either String)’
        arising from a use of ‘<|>’
    • In a stmt of a 'do' block:
        parseKeywords url' <|> (uriToptions =<< parseURIStr url')
      In the expression:
        do url' <- maybe (Left "failed to parse as string") Right
                     $ parseString url
           parseKeywords url' <|> (uriToptions =<< parseURIStr url')
      In an equation for ‘parseConnectionString’:
          parseConnectionString url
            = do url' <- maybe (Left "failed to parse as string") Right
                           $ parseString url
                 parseKeywords url' <|> (uriToptions =<< parseURIStr url')
    |
204 |   parseKeywords url' <|> (uriToptions =<< parseURIStr url')
    |                      ^^^
make: *** [Makefile:5: build] Error 1
```

Do you think a minor release is warranted?

--- a/src/Database/PostgreSQL/Simple/Options.hs
+++ b/src/Database/PostgreSQL/Simple/Options.hs
@@ -201,4 +201,7 @@ parseKeywords x = fmap mconcat . mapM (uncurry keywordToptions <=< toTuple . spl
 parseConnectionString :: String -> Either String Options
 parseConnectionString url = do
   url' <- maybe (Left "failed to parse as string") Right $ parseString url
-  parseKeywords url' <|> (uriToptions =<< parseURIStr url')
+  or (parseKeywords url') (uriToptions =<< parseURIStr url')
+  where
+    or (Left _) n = n
+    or m        _ = m 

